javaslang/src/test/java/javaslang/collection/euler/Euler26Test.java

```diff
@@ -0,0 +1,123 @@
+ * in its decimal fraction part. <p>
+ * See also
+ * <a href="https://projecteuler.net/problem=26">projecteuler.net problem 26
+ * </a>.
+ */
+public class Euler26Test {
+
+    @Test
+    public void shouldSolveProblem26() {
+        assertThat(recurringCycleLengthForDivisionOf1(2)._2).isEqualTo(0);
+        assertThat(recurringCycleLengthForDivisionOf1(3)._2).isEqualTo(1);
+        assertThat(recurringCycleLengthForDivisionOf1(4)._2).isEqualTo(0);
+        assertThat(recurringCycleLengthForDivisionOf1(5)._2).isEqualTo(0);
+        assertThat(recurringCycleLengthForDivisionOf1(6)._2).isEqualTo(1);
+        assertThat(recurringCycleLengthForDivisionOf1(7)._2).isEqualTo(6);
+        assertThat(recurringCycleLengthForDivisionOf1(8)._2).isEqualTo(0);
+        assertThat(recurringCycleLengthForDivisionOf1(9)._2).isEqualTo(1);
+        assertThat(recurringCycleLengthForDivisionOf1(10)._2).isEqualTo(0);
+        assertThat(deonominatorBelow1000WithTheLongetsRecurringCycleOfDecimalFractions()).isEqualTo(983);
+    }
+
+    private static int deonominatorBelow1000WithTheLongetsRecurringCycleOfDecimalFractions() {
+        return List.range(2, 1000)
+                .map(Euler26Test::recurringCycleLengthForDivisionOf1)
+                .maxBy(Tuple2::_2)
+                .get()._1;
+    }
+
+    private static Tuple2<Integer, Integer> recurringCycleLengthForDivisionOf1(int divisor) {
+        return Tuple.of(
+                divisor,
+                recurringCycleLengthInDecimalFractionPart(
+                        CharSeq.of(BigDecimal.ONE.divide(BigDecimal.valueOf(divisor), 2000, RoundingMode.UP).toString())
+                        .transform(removeLeadingZeroAndDecimalPoint())
+                        .transform(removeRoundingDigit())
+                        .transform(removeTrailingZeroes())
+                        .mkString()
+                ));
+    }
+
+    private static int recurringCycleLengthInDecimalFractionPart(String decimalFractionPart) {
+        return CharSeq.of(decimalFractionPart)
+                .reverse()
+                .toStream() // Stream is lazy evaluated which ensures the rest is only evaluated until the recurring cycle is found.
+                .transform(createCandidateCycles())
+                .transform(removeCandidatesLongerThanHalfTheFullString(decimalFractionPart))
+                .transform(findFirstRecurringCycle(decimalFractionPart))
+                .map(String::length)
+                .getOrElse(0);
+    }
+
+    private static Function1<CharSeq, CharSeq> removeLeadingZeroAndDecimalPoint() {
+        return seq -> seq.drop(2);
+    }
+
+    private static Function1<CharSeq, CharSeq> removeRoundingDigit() {
+        return seq -> seq.dropRight(1);
+    }
+
+    private static Function1<CharSeq, CharSeq> removeTrailingZeroes() {
+        return seq -> seq
+                .reverse()
+                .dropWhile(c -> c == '0') //Remove any trailing zeroes
+                .reverse();
```

Comment:
This rather clunky reversing and re-reversing is because there is no CharSeq#dropRightWhile. Should we add that (and dropRightUntil), or does adding that to CharSeq mean it should be added to other collections as well where it might not be a good method to use?